Extensions/Content/Dnn.PersonaBar.Extensions/Components/TaskScheduler/TaskSchedulerController.cs

```diff
@@ -167,5 +168,43 @@ public IEnumerable<ScheduleItem> GetScheduleItems(bool? enabled, string serverNa
                 return null;
             }
         }
+
+        /// <summary>
+        /// Gets list of servers to be recommended for the particular scheduler.
+        /// </summary>
+        /// <param name="schedulerId">Scheduler Id</param>
+        /// <returns>List of recommended servers</returns>
+        public IEnumerable<string> GetRecommendedServers(int schedulerId)
+        {
+            var recommendedServers = new List<string>();
+
+            var schedulersToRunOnSameWebServer = HostController.Instance.GetString(
+                "SchedulersToRunOnSameWebServer", 
+                string.Empty);
+
+            var schedulers = schedulersToRunOnSameWebServer.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (!schedulers.Any() || !schedulers.Select(x => x.Trim()).Contains(schedulerId.ToString()))
+            {
+                return recommendedServers;
+            }
+
+            foreach (var scheduler in schedulers)
+            {
+                if (!int.TryParse(scheduler.Trim(), out int id) || id == schedulerId)
+                {
+                    continue;
+                }
+
+                var schedule = SchedulingProvider.Instance().GetSchedule(id);
+
+                if (schedule != null && schedule.Enabled && !string.IsNullOrWhiteSpace(schedule.Servers))
+                {
+                    recommendedServers.AddRange(schedule.Servers.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries));
```

Comment:
You are not trimming before distinct! If your server list holds values like in the below you may run into trouble:
`[  "1  ,   2,   x,y   \r\n....",  "1,2,x,y" ]`

To overcome such issues you need fewer loops and more Linq as readability will be greater so you can notice such issues easily:

```
        public IEnumerable<string> GetRecommendedServers(int schedulerId)
        {
            var schedulersToRunOnSameWebServer = HostController.Instance.GetString(
                "SchedulersToRunOnSameWebServer",
                string.Empty);

            var schedulers = schedulersToRunOnSameWebServer
                .Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(scheduler => scheduler.Trim());

            if (!schedulers.Contains(schedulerId.ToString()))
            {
                return Enumerable.Empty<string>();
            }
            else
            {
                var scheduleServerLists = schedulers
                    .Select(scheduler => int.TryParse(scheduler, out int id) ? id : (int?)null)
                    .Where(id => id != null && id != schedulerId)
                    .Select(id => SchedulingProvider.Instance().GetSchedule((int)id))
                    .Where(schedule => schedule != null && schedule.Enabled && !string.IsNullOrWhiteSpace(schedule.Servers));

                return scheduleServerLists
                    .SelectMany(schedule => schedule.Servers.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
                    // You are missing .Select(x => x.Trim()) before Distinct
                    .Distinct()
                    .OrderBy(x => x);
            }
        }
```

At least apply trim here or convert to like in the recommendation below,
With trim: `recommendedServers.AddRange(schedule.Servers.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries)).Select(server => server.Trim())`